lib/tasks/fix_nil_ethnicity_arrays.rake

```diff
@@ -1,39 +1,50 @@
 # Run this with RAILS_ENV=production bundle exec rake migrations:fix_nil_ethnicity_arrays application_hbx_id="12345"
 # application_hbx_id is not required but can be used for testing/fixing on a single application.
 # By default, rake will run on all ATP-ingested applications in a draft state
+# To run just a report: RAILS_ENV=production bundle exec rake migrations:fix_nil_ethnicity_arrays report_only=true
 namespace :migrations do
   desc "Remove nils from person and applicant ethnicity arrays"
   task :fix_nil_ethnicity_arrays => :environment do |_task, _args|
     # - find draft applications that were ingested via ATP (or using provided hbx_id)
-    # - check and update person ethnicity array
-    # - check and update applicant ethnicity array
+    # - check and update person ethnicity arrays
+    # - check and update applicant ethnicity arrays
 
     def find_person(person_hbx_id)
       Person.find_by(hbx_id: person_hbx_id)
     rescue StandardError => e
-      puts "Error finding person document - hbx_id: #{person_hbx_id}"
+      ["Error finding person document - hbx_id: #{person_hbx_id}"]
     end
 
-    def fix_person_ethnicity_array(ethnicities, _application, person)
-      Person.skip_callback(:update, :after, :person_create_or_update_handler)
-      person.ethnicity = ethnicities.compact
-      updated = person.save(validate: false)
-      Person.set_callback(:update, :after, :person_create_or_update_handler)
-      raise StandardError, "Failed to save person (hbx_id): #{person.hbx_id}" unless updated
+    def fix_person_ethnicity_array(ethnicities, application, person)
+      primary = application.primary_applicant ? application.primary_applicant.person_hbx_id : "no_primary_found"
+      result = [application.hbx_id, primary, person.hbx_id, person.ethnicity.to_s]
+      if ENV['report_only'].nil?
+        Person.skip_callback(:update, :after, :person_create_or_update_handler)
+        person.ethnicity = ethnicities.compact
+        updated = person.save(validate: false)
+        Person.set_callback(:update, :after, :person_create_or_update_handler)
+        result << person.ethnicity.to_s
+        raise StandardError, "Failed to save person (hbx_id): #{person.hbx_id}" unless updated
+      end
+      result
     rescue StandardError => e
-      puts "Error updating peson ethnicity array - #{e}"
+      ["Error updating person ethnicity array - #{e}"]
```

Comment:
why this change?

Reply:
This was meant to log any encountered errors in the output .csv file.  I modified this so that errors are printed to the console and only successful updates are logged in the .csv